cmd/main.go

```diff
@@ -12,13 +14,21 @@ func main() {
 	config := internal.NewGlobalConfig()
 
 	// Setup logger
-	log := internal.NewDefaultLogger()
+	log := log2.NewDefaultLogger(config.LogLevel, config.LogFormat)
 
 	// Perform config validation
 	config.Validate()
 
+	// Query the OIDC provider
+	config.SetOidcProvider()
+
 	// Build server
-	server := internal.NewServer()
+	stopChannel := make(chan struct{})
```

Comment:
How is this new channel being used? I understand that this function will close the channel on exit, but we never wait for workers to complete their work. Since the `main` function is closing this channel on exit, what is the reason for having this channel?

Reply:
The channel is eventually passed to the informer cache loop. When the channel is closed, it signals that the informer should be shutdown.

Reply:
Right, but we never wait for cache informer loop to stop the work, but rather exit the program immediately. So there is not much value in having this channel here, if we don't act on it.

Reply:
Thanks for catching this. I was thinking ListenAndServe returned gracefully on SIGTERM and that the deferred `close` would run. From what I am reading, the informer will shutdown when the connection drops. Though, I am considering implementing a signal catch to perform a graceful shutdown of http workers; which would also be a good time to utilize the stop channel to explicitly shutdown the informer. 

I am refactoring some other things, so I've set this back to WIP. Thanks again for your review thus far, I really appreciate it.